### ISSUE TITLE
Oracle lowercased alias with double quotes not repsected by rule L059

### ISSUE THREAD

**Issue by kkaiser** [sqlfluff/sqlfluff#3704]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running sqlfluff (version 1.2.1) it wants to fix double quotes on a column alias.

```bash
sqlfluff lint --dialect oracle test.sql 
== [test.sql] FAIL                                                                                    
L:   1 | P:  15 | L059 | Unnecessary quoted identifier "renamed_col".
All Finished 📜 🎉!
```

Content of test.sql
```sql
SELECT col AS "renamedCol" 
FROM table
```

In my opinion this is a false positive trigger for the oracle dialect.

### Expected Behaviour

I expect the rule not to be triggered since in Oracle double quotes are required if an alias contains spaces *OR* if it contains lower-case chars.

### Observed Behaviour

Using `sqlfluff fix` causes the query to return a bad result for the column name.

### How to reproduce

`pip install sqlfluff` and run above commands

### Dialect

oracle

### Version

1.2.1

### Configuration

n/a see CLI options

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
> I expect the rule not to be triggered since in Oracle double quotes are required if an alias contains spaces OR if it contains lower-case chars.

My understanding is that it's not so much that double quotes are required when lowercase characters are used, but that using them ensures case sensitivity.

So this is valid if I understand correctly:

```sql
SELECT col1 AS Column1
WHERE column1 = 'test'
```

whereas this is not correct:


```sql
SELECT col1 AS "Column1"
WHERE column1 = 'test'
```

So is using case sensitive column names a bit of an anti-pattern? And so is it right for SQLFluff to flag this and suggest removing them? And if you really want to use them to disable L059 in those cases?

**Comment by kkaiser** [comment]
This is a valid query:
![image](https://user-images.githubusercontent.com/1774205/182360570-0b48cd0a-7ca1-4c7b-ae5f-9e1484488660.png)

But it returns an unexpected result (compared to other dialects). Only with the quotes it returns the desired result:
![image](https://user-images.githubusercontent.com/1774205/182360799-0744cadc-04fc-43b2-94fb-78545ec2d4f0.png)

In my opinion `sqlfluff fix` should not break the query. Removing the quotes changes the casing of the column name that is desired.

The "AS" syntax to generate aliases doesn't rely have another purpose then mapping column names, in my opinion casing is part of that.

It's true that I can ignore the rule but I just wanted to hear your opinion and make sure you really consider this an anti-pattern.

**Comment by tunetheweb** [comment]
Well oracle is case insensitive by default, and you need to explicitly add the quotes to change the default. This indicates to me what they consider best practice and matches my own thoughts to be honest. There is no _need_ to make the column case sensitive (at least within Oracle but understand why you might want it for using the results outside of Oracle.

How do you think SQLFluff should handle this? Should it automatically turn off rule L059 for Oracle? Or should we ignore identifiers (or just aliases?) with mixed cases from this rule (like we do for spaces)? Or something else.

**Comment by kkaiser** [comment]
I absolutely agree that there is no need to do it.

In my opinion (which is not necessarily based on tons of experience what
practices other linters follow), I would prefer that aliases (just aliases)
with mixed cases are ignored from the rule. Since the quotes are (most
likely) set there intentionally, if the string is all uppercase then the
quotes are in fact just extra fluff :)

**Comment by saulotoledo** [comment]
@peterbud, @keraion, @tunetheweb, who should decide the next steps on this issue?
